addons/l10n_latam_check/wizards/account_payment_register.py

```diff
@@ -0,0 +1,63 @@
+from odoo import models, fields, api
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    l10n_latam_check_id = fields.Many2one(
+        comodel_name='account.payment',
+        string='Check',
+    )
+    l10n_latam_check_bank_id = fields.Many2one(
+        comodel_name='res.bank',
+        string='Check Bank',
+        compute='_compute_l10n_latam_check_data', store=True, readonly=False,
+    )
+    l10n_latam_check_issuer_vat = fields.Char(
+        string='Check Issuer VAT',
+        compute='_compute_l10n_latam_check_data', store=True, readonly=False,
+    )
+    l10n_latam_check_number = fields.Char(
+        string="Check Number",
+    )
+    l10n_latam_manual_checks = fields.Boolean(
+        related='journal_id.l10n_latam_manual_checks',
+    )
+    l10n_latam_check_payment_date = fields.Date(
+        string='Check Cash-In Date', help="Date from when you can cash in the check, turn the check into cash",
+    )
+
+    @api.depends('payment_method_line_id.code', 'partner_id')
+    def _compute_l10n_latam_check_data(self):
+        new_third_party_checks = self.filtered(lambda x: x.payment_method_line_id.code == 'new_third_party_checks')
+        (self - new_third_party_checks).update({'l10n_latam_check_bank_id': False, 'l10n_latam_check_issuer_vat': False})
+        for rec in new_third_party_checks:
+            rec.l10n_latam_check_bank_id = rec.partner_id.bank_ids[:1].bank_id
+            rec.l10n_latam_check_issuer_vat = rec.partner_id.vat
+
+    @api.onchange('l10n_latam_check_id')
+    def _onchange_amount(self):
+        for rec in self.filtered('l10n_latam_check_id'):
+            rec.amount = rec.l10n_latam_check_id.amount
+
+    def _create_payment_vals_from_wizard(self, batch_result):
+        vals = super()._create_payment_vals_from_wizard(batch_result)
+        vals.update({
+            'l10n_latam_check_id': self.l10n_latam_check_id.id,
+            'l10n_latam_check_bank_id': self.l10n_latam_check_bank_id.id,
+            'l10n_latam_check_issuer_vat': self.l10n_latam_check_issuer_vat,
+            'check_number': self.l10n_latam_check_number,
+            'l10n_latam_check_payment_date': self.l10n_latam_check_payment_date,
+        })
+        return vals
+
+    @api.onchange('l10n_latam_check_number')
+    def _onchange_l10n_latam_check_number(self):
+        for rec in self.filtered(lambda x: x.journal_id.company_id.country_id.code == "AR" and x.l10n_latam_check_number
+                                 and x.l10n_latam_check_number.isdecimal()):
+            rec.l10n_latam_check_number = '%08d' % int(rec.l10n_latam_check_number)
```

Comment:
@william-andre 
In this case, is because we only want to suggest the bumper padding and format if the user has only added numbers (without any letters, spaces, dashes etc). This method is an auto suggestion in the order they use the same padding and numbering for the checks. If they use something different we do not do the re-formatting

Please let me know if this resolve your concerns

Reply:
we add a dosctring
@zaoral

Reply:
@william-andre @jjscarafia @zaoral can we close this comment?
cc' @jco-odoo

Reply:
Hi @william-andre 
Not sure if the explanation resolves the doubts, or if you have any suggestions.
Please let us know so we can close the pending comment on this pr.

Best Regards

Reply:
Ok, will probably need to be move to `l10n_ar` when the dependence will be set (in `master`)